### PR TITLE
Inline Twine.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Thought exercise in making twine an inline dependency rather than a git submodule.
Not the most minimal version of twine possible, just a working build of it.

The other option would be to make twine a real package on hackage and depend on that.

I have made a start on removing the submodules in twine here https://github.com/tmcgilchrist/twine/tree/topic/either 

The remaining issue is the use of `P` and whether that is going up on hackage or just inline a minimal version.

### CODE REVIEW

**Comment by erikd-ambiata** [comment]
I'm cool with this. I think its important to get mafia to a stage where people can just cabal install it.

As for `p` I think mafia should just use the standard prelude and the stuff in `x` does mafia use anything that is not in `transformers-either`? Finally, `disorder` should just be swapped out for hedgehog.

👍

File: src/Mafia/Twine/Async.hs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import           Mafia.Twine.Snooze
+
+import           System.IO (IO)
+
+import           Control.Monad.Trans.Either
+
+data AsyncTimeout =
+  AsyncTimeout Duration
+  deriving (Eq, Show)
```
Comment (erikd) [inline src/Mafia/Twine/Async.hs:30]: Any reason for that not to be a `newtype`?

File: .gitmodules [hunk 2]
```diff
@@ -1,12 +1,9 @@
...
 	url = https://github.com/ambiata/disorder.hs
 [submodule "lib/x"]
 	path = lib/x
 	url = https://github.com/ambiata/x
-[submodule "lib/twine"]
-	path = lib/twine
-	url = https://github.com/ambiata/twine
+[submodule "lib/p"]
+	path = lib/p
+	url = git@github.com:ambiata/p.git
```
Comment (erikd) [inline .gitmodules]: P still needed as a submodule? What about X?

File: src/Mafia/Twine/Async.hs (hunk 1 above)
Comment (tmcgilchrist) [reply src/Mafia/Twine/Async.hs:30]: I'm purposefully not improving / changing twine beyond what is absolutely necessary to get mafia working. There's a good case for either making twine a real thing on hackage or improving the twine inlined here (I'd be voting for the first option).

File: .gitmodules (hunk 2 above)
Comment (tmcgilchrist) [reply .gitmodules]: There is still work in progress to remove all the submodules, first with x-optparse and then anything in disorder. Only then can i remove p entirely.

Comment (erikd-ambiata) [reply .gitmodules]: So probably should remove all these submodules in one go.

**Review by erikd — APPROVED** [review]

**Pull request merged**